crates/gfx-plugin/src/rendering.rs

```diff
@@ -0,0 +1,225 @@
+pub struct Position {
+    /// The vector-representation of the position.
+    pub vector: Vector3<f32>,
+}
+
+impl Default for Position {
+    fn default() -> Self {
+        Self {
+            vector: Vector3::zero(),
+        }
+    }
+}
+
+/// An orientation in 3D-space.
+#[derive(Debug, Copy, Clone)]
+pub struct Rotation {
+    /// Rotation represented as a quaternion.
+    pub quaternion: Quaternion<f32>,
+}
+
+impl Default for Rotation {
+    fn default() -> Self {
+        Self {
+            quaternion: Quaternion::one(),
+        }
+    }
+}
+
+/// At which scale to render the object.
+///
+/// Each axis can be scaled differently to stretch the object.
+#[derive(Debug, Copy, Clone)]
+pub struct Scale {
+    /// The vector-representation of the scale.
+    pub vector: Vector3<f32>,
+}
+
+impl Default for Scale {
+    fn default() -> Self {
+        Self {
+            vector: Vector3::new(1.0, 1.0, 1.0),
+        }
+    }
+}
+
+pub(crate) type RenderQuery<'parameters> = Query<
+    'parameters,
+    (
+        Read<'parameters, Model>,
+        Read<'parameters, Position>,
+        Read<'parameters, Rotation>,
+        Read<'parameters, Scale>,
+    ),
+>;
+
+pub(crate) type RenderData = Vec<(ModelHandle, Vec<Transform>)>;
+
+pub(crate) fn rendering_system(mut render_data_sender: RingSender<RenderData>, query: RenderQuery) {
+    let grouped_transforms: Vec<_> = query
+        .into_iter()
+        .map(|(model, position, rotation, scale)| {
+            (
+                model,
+                Transform {
+                    position: position.vector,
+                    rotation: rotation.quaternion,
+                    scale: scale.vector,
+                },
+            )
+        })
+        .group_by(|(model, _)| model.handle)
+        .into_iter()
+        .map(|(model, group)| {
+            let (_, transforms): (Vec<_>, Vec<_>) = group.unzip();
+            (model, transforms)
+        })
+        .collect();
+
+    render_data_sender
+        .send(grouped_transforms)
+        .expect("this system should have stopped running when the receiver is dropped");
+}
+
+/// An error in construction of a rendered entity.
+#[derive(Error, Debug)]
+pub enum RenderedEntityBuilderError {
+    /// Failed to create a new entity.
+    #[error("failed to create a new entity")]
+    EntityCreation(#[source] Box<dyn Error + Send + Sync>),
+    /// Failed to add component to entity.
+    #[error("failed to add component to entity")]
+    ComponentAdding(#[source] Box<dyn Error + Send + Sync>),
+}
+
+/// Whether an operation on the graphical application succeeded.
+pub type RenderedEntityBuilderResult<T, E = RenderedEntityBuilderError> = Result<T, E>;
+
+/// Builds a visible (i.e. rendered) entity.
+#[derive(Debug)]
+pub struct RenderedEntityBuilder<'app, App> {
+    application: &'app mut App,
+    entity: Entity,
+    model: Model,
+    position: Option<Position>,
+    rotation: Option<Rotation>,
+    scale: Option<Scale>,
```

Comment:
Yeah, that's an interesting idea. We did consider making it possible to use this builder to add any type of component, but we realized that it could lead to confusion when users try to specify one of the rendering-components using `builder.add_component` rather than `builder.with_position`, because then the default position would overwrite the one specified. 

We tried making a method on `Application` that could add multiple components at the same time, but could not do so in a satisfactory way as a vector of objects needs them to be uniform (i.e. not different component types). You could box all the components but at that point you've already lost so much in ergonomics it's not worth it. But maybe defining prefabs using a trait and a struct could be a way around this.

If you want to you could create an issue for this